server/tests/utils/test_weights.py

```diff
@@ -0,0 +1,480 @@
+import torch
+from text_generation_server.utils.weights import Weights
+from text_generation_server.layers.gptq import GPTQWeight
+from text_generation_server.layers.exl2 import Exl2Weight
+from text_generation_server.layers.marlin import MarlinWeight
+from types import SimpleNamespace
+from typing import List, Optional, Dict, Union
+from pathlib import Path
+
+dummy_file_system = {
+    "test_weights": {
+        "layer.0.weight": torch.tensor(
+            [
+                [1, 2],
+                [3, 4],
+            ],
+            dtype=torch.float32,
+        ),
+    },
+    "test_weights_2": {
+        "layer.1337.weight": torch.tensor(
+            [
+                [1, 2, 3, 4],
+                [5, 6, 7, 8],
+            ],
+            dtype=torch.float32,
+        ),
+    },
+    "test_get_multi_weights_col_packed": {
+        "col_packed.weight": torch.tensor(
+            [
+                [1, 2],
+                [3, 4],
+                [5, 6],
+                [7, 8],
+            ],
+            dtype=torch.float32,
+        ),
+        "col_packed_2.weight": torch.tensor(
+            [
+                [1, 2],
+                [3, 4],
+                [5, 6],
+                [7, 8],
+            ],
+            dtype=torch.float32,
+        ),
+    },
+    "test_get_multi_weights_row": {
+        "row_packed.weight": torch.tensor(
+            [
+                [1, 2],
+                [3, 4],
+                [5, 6],
+                [7, 8],
+            ],
+            dtype=torch.float32,
+        ),
+    },
+    "test_get_multi_weights_row_gptq": {
+        "weight.qweight": torch.tensor(
```

Comment:
I think the quantizers should probably be tested with the real data types to check that there are no accidental dtype conversions happening. So `int32` for `qweight`, `qzeros`, and `g_idx`, and `f16` for `scales`.

Maybe also real shapes? (e.g. currently `qzeros` does not line up with just having one group) Probably less important.

Reply:
great points, I've updated the types in the latest changes but the values are still not fully realistic, happy to follow up in this or a future PR

Reply:
also I'm not 100% sure if `col_packed_exl2` is suppose to be supported or not? I was a bit confused about the expected weight names ("weights" vs "qweights")... is this intended? I'll update the test accordingly 🙂

Reply:
Great question! This path is missing in `weights`, I'll create a ticket and look it. I can update the test here once it's properly implemented.